silica/compile.py

```diff
@@ -168,17 +169,29 @@ def compile(coroutine, file_name=None, mux_strategy="one-hot", output='verilog',
     num_states = len(states)
     if has_ce:
         raise NotImplementedError("add ce to module decl")
+
+    # declare module and ports
+    module = vg.Module(module_name)
+    for o in outputs:
+        module.Output(o, width_table.get(o, 1))
+    if coroutine._inputs:
+        for i,t in coroutine._inputs.items():
+            if isinstance(t, m.BitKind):
+                module.Input(i)
+            else:
+                module.Input(i, t.N)
+    module.Input("CLK")
     verilog_source += f"""
 module {module_name} ({io_string}, input CLK);
 """
```

Comment:
We should be able to get rid of all this old `verilog_source` related code, correct?

Reply:
I'm leaving it in for now because some of the old code paths depend on variables that were created and I don't have full coverage of the old paths yet, just enough to pass the existing tests. It helps a lot to be able to dump both strings at any point during codegen and double check the output is as intended. 

Once I add those remaining cases and refactor a bit I'll delete this stuff.

Reply:
Got it, let's not remove it for now then.